contract/src/main.rs

```diff
@@ -499,90 +516,65 @@ pub extern "C" fn mint() {
 
     let owned_tokens_item_key = utils::get_owned_tokens_dictionary_item_key(token_owner_key);
 
-    let owned_tokens_actual_key = Key::dictionary(
-        utils::get_uref(
-            OWNED_TOKENS,
-            NFTCoreError::MissingOwnedTokens,
-            NFTCoreError::InvalidOwnedTokens,
-        ),
-        owned_tokens_item_key.as_bytes(),
-    );
-
-    // Update owned tokens dictionary
-    let maybe_owned_tokens: Option<Vec<TokenIdentifier>> = {
-        match identifier_mode {
-            NFTIdentifierMode::Ordinal => utils::get_dictionary_value_from_key::<Vec<u64>>(
-                OWNED_TOKENS,
-                &owned_tokens_item_key,
-            )
-            .map(|token_indices| {
-                token_indices
-                    .into_iter()
-                    .map(TokenIdentifier::new_index)
-                    .collect()
-            }),
-            NFTIdentifierMode::Hash => utils::get_dictionary_value_from_key::<Vec<String>>(
-                OWNED_TOKENS,
-                &owned_tokens_item_key,
-            )
-            .map(|token_hashes| {
-                token_hashes
-                    .into_iter()
-                    .map(TokenIdentifier::new_hash)
-                    .collect()
-            }),
-        }
-    };
+    // Update the forward and reverse trackers
+    utils::update_forward_and_reverse_token_trackers(
+        current_number_of_minted_tokens,
+        token_identifier.clone(),
+    );
 
-    // Update the value in the owned_tokens dictionary.
-    match maybe_owned_tokens {
-        Some(mut owned_tokens) => {
-            // Check that we are not minting a duplicate token.
-            if owned_tokens.contains(&token_identifier) {
-                runtime::revert(NFTCoreError::FatalTokenIdDuplication);
-            }
+    let current_page_number = current_number_of_minted_tokens / PAGE_SIZE;
+    let current_page_index = current_number_of_minted_tokens % PAGE_SIZE;
```

Comment:
`page_position` 

(this will be the new token's address)

Reply:
Renamed variable